File: wheystore/wheystore.Web/Modules/SanPhamDB/MaGiamGia/MaGiamGiaColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.MaGiamGia")]
[BasedOnRow(typeof(MaGiamGiaRow), CheckNames = true)]
public class MaGiamGiaColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string MaCode { get; set; }
    public string MoTa { get; set; }
    public short LoaiGiam { get; set; }
    public decimal GiaTri { get; set; }
    public decimal DonToiThieu { get; set; }
    public DateTime NgayBatDau { get; set; }
    public DateTime NgayKetThuc { get; set; }
    public int SoLanDungToiDa { get; set; }
    public int SoLanDungMoiKhach { get; set; }
    public bool IsActive { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/MaGiamGiaColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MaGiamGiaRow } from "./MaGiamGiaRow";

export interface MaGiamGiaColumns {
    Id: Column<MaGiamGiaRow>;
    MaCode: Column<MaGiamGiaRow>;
    MoTa: Column<MaGiamGiaRow>;
    LoaiGiam: Column<MaGiamGiaRow>;
    GiaTri: Column<MaGiamGiaRow>;
    DonToiThieu: Column<MaGiamGiaRow>;
    NgayBatDau: Column<MaGiamGiaRow>;
    NgayKetThuc: Column<MaGiamGiaRow>;
    SoLanDungToiDa: Column<MaGiamGiaRow>;
    SoLanDungMoiKhach: Column<MaGiamGiaRow>;
    IsActive: Column<MaGiamGiaRow>;
}

export class MaGiamGiaColumns extends ColumnsBase<MaGiamGiaRow> {
    static readonly columnsKey = 'SanPhamDB.MaGiamGia';
    static readonly Fields = fieldsProxy<MaGiamGiaColumns>();
}